src/app/services/cross-table.service.ts

```diff
@@ -181,6 +183,7 @@ export class CrossTableService {
       try {
         this.crossTable.rowHeaderConstraints = this.crossConstraints(this.rowConstraints);
         this.crossTable.columnHeaderConstraints = this.crossConstraints(this.columnConstraints);
+        this.crossTable.constraint = this.constraintService.cohortSelectionConstraint;
```

Comment:
Shouldn't this be the combination of all selected cohorts? And then also for exports? Looks like some of that code (in cohort service could use some refactoring...)